### ISSUE TITLE
Node distance calculation wrong

### ISSUE THREAD

**Issue by cyberlarino** [Skretzo/shortest-path#10]
Current pathfinding implementation is somewhat wrong:
Currently considers the past path as the node with least cardinal distance from the target.

This is wrong. In comparison to other pathfinding algorithms:
- **Dijkstra:** Evaluate the next node with lowest cost, _**based on cost it takes to get to that node**_
- **A\*:** Evaluate the next node with lowest cost, based on the cost it takes to get to that node + estimated distance to target

In conclusion, current distance/cost estimation doesn't properly take account for cost/distance to each node. This is maybe fine for traversing grids, but if teleports/transport methods are taken into account it will behave suboptimally.

**Comment by Skretzo** [comment]
This plugin aims to create paths that are as accurate as possible to the in-game OSRS pathfinding. For that reason, it is using the Breadth-first search algorithm. Dijkstra is fairly similar to BFS, but does not offer any benefits over the current BFS algorithm and its modifications. A* could probably speed up the pathfinding, but it would most likely be very hard or impossible to get an accurate OSRS path with it.
